File: server/plugin/tools/router/environment.go
package router

import (
	"github.com/gin-gonic/gin"
)

type environment struct{}

func (r *environment) Init(public, private *gin.RouterGroup) {
	group := private.Group("toolsEnvironment")
	group.POST("createEnvironment", apiInfo.Environment.CreateEnvironment)
	group.PUT("updateEnvironment", apiInfo.Environment.UpdateEnvironment)
	group.DELETE("deleteEnvironment", apiInfo.Environment.DeleteEnvironment)
	group.GET("findEnvironment", apiInfo.Environment.FindEnvironment)
	group.GET("getEnvironmentList", apiInfo.Environment.GetEnvironmentList)
}

File: web/src/plugin/tools/api/environment.js
import service from '@/utils/request'

export const createEnvironment = (data) => {
  return service({
    url: '/toolsEnvironment/createEnvironment',
    method: 'post',
    data
  })
}

export const updateEnvironment = (data) => {
  return service({
    url: '/toolsEnvironment/updateEnvironment',
    method: 'put',
    data
  })
}

export const deleteEnvironment = (params) => {
  return service({
    url: '/toolsEnvironment/deleteEnvironment',
    method: 'delete',
    params
  })
}

export const findEnvironment = (params) => {
  return service({
    url: '/toolsEnvironment/findEnvironment',
    method: 'get',
    params
  })
}

export const getEnvironmentList = (params) => {
  return service({
    url: '/toolsEnvironment/getEnvironmentList',
    method: 'get',
    params
  })
}
